library/src/validations/email/email.ts

```diff
@@ -4,15 +4,13 @@ import { getOutput, getPipeIssues } from '../../utils/index.ts';
 /**
  * Creates a validation functions that validates a email.
  *
- * Hint: The regex used is not perfect, but should work for most emails.
- *
  * @param error The error message.
  *
  * @returns A validation function.
  */
 export function email<TInput extends string>(error?: ErrorMessage) {
   return (input: TInput): PipeResult<TInput> =>
-    !/^(([^<>()[\].,;:\s@"]+(\.[^<>()[\].,;:\s@"]+)*)|(".+"))@(([^<>()[\].,;:\s@"]+\.)+[^<>()[\].,;:\s@"]{2,})$/i.test(
+    !/^(“|[A-Z0-9_+-]+\.?)*[A-Z0-9_+-]”?@(([A-Z0-9][A-Z0-9-]*\.)+[A-Z]{2,}|([0-9]{3}\.[0-9]{3}\.[0-9]{3}\.[0-9]{3}))$/i.test(
```

Comment:
Why do we need `“` and `”` in the new regex?

Reply:
@fabian-hiller 
Thank you for checking!

> Why do we need “ and ” in the new regex?

That's because there is already the test which enclosing with `“ and ” ` is valid
https://github.com/fabian-hiller/valibot/pull/180/files#diff-5749fd906e956e45be3645c6d1f4a32b5872bd88160a7364b775c6feb0f2d61eR17

Should I delete `“ and ”` from the regex, and also the test above?

Reply:
Thanks for the hint! We should check if this type of email is supported.

Reply:
Just noting `“` is definitely not the same as `"`, so not sure if the regex is intended to be like that